language/stdlib/modules/Errors.move

```diff
@@ -14,15 +14,17 @@ address 0x1 {
 ///     framework evolves. TODO(wrwg): determine what kind of stability guarantees we give about reasons/
 ///     associated module.
 module Errors {
+    spec module { pragma verify; }
 
     /// A function to create an error from from a category and a reason.
     fun make(category: u8, reason: u64): u64 {
         (category as u64) + (reason << 8)
     }
     spec fun make {
         pragma opaque = true;
-        aborts_if false;
-        ensures result == category + (reason << 8);
+        aborts_if category + (reason << 8) > max_u64();
```

Comment:
Since the abstract version cannot abort, I would suggest to add `aborts_if [abstract] false` and mark this as concrete. Gives on thing less to reason about for the prover.

I'm not sure 100% though whether abstract/concrete works correctly for aborts_if, not remember have tested it. In case you run into trouble with this, perhaps just a TODO and verify = false. That this is light weight for callers is the most important matter.

Reply:
BTW, why is this even needed? The expression can never overflow (shift does not overflow at least in Move. Perhaps there is a bug in our implementation of it.)

Reply:
If left shift is implemented simply as multiplication with a constant without any guard, I suppose it could lead to a number larger than max_u64.